mtp_api/apps/security/serializers.py

```diff
@@ -220,9 +224,92 @@ def update(self, instance, validated_data):
         return super().update(instance, validated_data)
 
 
+class CheckAutoAcceptRuleStateSerializer(serializers.ModelSerializer):
+    active = serializers.BooleanField(required=False)
```

Comment:
why is this not required? is it just to allow this call to omit the `active` param?
https://github.com/ministryofjustice/money-to-prisoners-noms-ops/pull/510/files#diff-8ea59d9e31bda3ec134036535c9ff9709e0d74eb87794d9b602a87fb2314a638R345
might make the code and validation errors a bit clearer if on create it had to be True, but present always.

Reply:
see point just below that's related. i'm not saying this needs to change or anything.

how i understand it works now is:
- new auto-accept does require an `active` flag because the serialiser always uses True
- subsequent calls require an `active` flag
- but they do not validate whether it's the opposite of the latest flag

we could validate subsequent calls to ensure the flag cycles True/False, but i feel that's overkill. i'd probably have just made it simpler by making `active` always but overridden it to True on creation.

Reply:
`active` is not required on `create` but is required everywhere else. The view will use these rules for its validation, even those in the nested serializer, so the data structure received to the API needs to conform to these rules (to prove this. comment out this line and run the tests). I agree that it would be better to be explicit about the conditions when it is needed, I'll codify the above into a  `verify_active` method.

Agreed about not validating the flipping of state. I'll leave the actual mutation of the values as it is if you're happy with that

Reply:
i'm perfectly happy either way!